### ISSUE TITLE
No autocompletion for packages 

### ISSUE THREAD

**Issue by linkpy** [zigtools/zls#323]
File `pkg/pkg.zig` :
```zig
const std = @import("std");

pub fn hello() void {
    std.log.info("hello from pkg");
}
```

File `src/main.zig` :
```zig
const std = @import("std");
const pkg = @import("pkg");

pub fn main() anyerror!void {
    pkg.hello();
}
```

File `build.zig` :
```zig
const std = @import("std");

pub fn build(b: *std.build.Builder) void {
    const target = b.standardTargetOptions(.{});
    const mode = b.standardReleaseOptions();

    const exe = b.addExecutable("test0", "src/main.zig");
    exe.setTarget(target);
    exe.setBuildMode(mode);
    exe.addPackagePath("pkg", "pkg/pkg.zig");
    exe.install();

    const run_cmd = exe.run();
    run_cmd.step.dependOn(b.getInstallStep());
    if (b.args) |args| {
        run_cmd.addArgs(args);
    }

    const run_step = b.step("run", "Run the app");
    run_step.dependOn(&run_cmd.step);
}
```

When typing `pkg.` in the `main.zig` file, no autocompletion is given by ZLS.
We are using Zig and ZLS master.
We tried using `b.addPackage`, still not ok.
We tried deleting and recreating the zls configuration file, still not ok.

It used to work for a more complex project. That project was untouched for a few monthes and after updating zig+zls to master the package autocompletion was out in that project too.

**Comment by linkpy** [comment]
After getting some help on the Zig Discord : 

My ZLS was ill-configured and was missing the path to the `build_runner.zig` file. We set it up but that didn't solve the issue.
We are getting an error now : 

```
[debug-doc_store] Opened document: file:///home/pancake/Documents/devel/zig/test0/src/main.zig
[debug-doc_store] Opened document: file:///home/pancake/Documents/devel/zig/test0/build.zig
[debug-doc_store] Document is a build file, extracting packages...
[debug-doc_store] Failed to load packages of build file file:///home/pancake/Documents/devel/zig/test0/build.zig (error: error.RunFailed)
[debug-doc_store] Associated build file `file:///home/pancake/Documents/devel/zig/test0/build.zig` to document `file:///home/pancake/Documents/devel/zig/test0/src/main.zig`
[debug-doc_store] Opened document: file:///home/pancake/Documents/repos/zig/build/lib/zig/std/std.zig
```

We tried to run the build_runner itself which seems to work fine :

```
$ zig run /home/pancake/Documents/repos/zls/zig-out/bin/build_runner.zig --cache-dir ~/.cache/zls --pkg-begin @build@ ./build.zig --pkg-end
pkgpkg/pkg.zig
pkgpkg/pkg.zig
pkgpkg/pkg.zig
```

**Comment by alexnask** [comment]
For context, this is the only place error.RunFailed can be returned from: https://github.com/zigtools/zls/blob/master/src/document_store.zig#L137  

So it seems that ChildProcess.exec is returning a term which is not set to .ExitCode for some reason :/

**Comment by linkpy** [comment]
After some more investigation and adding more debug logging we found out what was the issue.
That execution of Zig when loading the packages terminates with a signal 6 (SIGABORT) due to a panic : 

```
thread 850503 panic: attempt to unwrap error: NotDir
/home/pancake/Documents/repos/zig/lib/std/process.zig:177:46: 0x55bb1e3e7501 in std.process.getEnvVarOwned (zig1)
        const result = os.getenv(key) orelse return error.EnvironmentVariableNotFound;
                                             ^
/home/pancake/Documents/repos/zig/lib/std/process.zig:177:46: 0x55bb1e3e7501 in std.process.getEnvVarOwned (zig1)
        const result = os.getenv(key) orelse return error.EnvironmentVariableNotFound;
                                             ^
/home/pancake/Documents/repos/zig/lib/std/process.zig:177:46: 0x55bb1e3e7501 in std.process.getEnvVarOwned (zig1)
        const result = os.getenv(key) orelse return error.EnvironmentVariableNotFound;
                                             ^
/home/pancake/Documents/repos/zig/lib/std/process.zig:177:46: 0x55bb1e3e7501 in std.process.getEnvVarOwned (zig1)
        const result = os.getenv(key) orelse return error.EnvironmentVariableNotFound;
                                             ^
/home/pancake/Documents/repos/zig/lib/std/process.zig:177:46: 0x55bb1e3e7501 in std.process.getEnvVarOwned (zig1)
        const result = os.getenv(key) orelse return error.EnvironmentVariableNotFound;
                                             ^
/home/pancake/Documents/repos/zig/lib/std/heap.zig:147:9: 0x55bb1e2e2318 in std.heap.CAllocator.resize (zig1)
        return error.OutOfMemory;
        ^
/home/pancake/Documents/repos/zig/lib/std/mem/Allocator.zig:315:16: 0x55bb1e5aa3cd in std.mem.Allocator.resize (zig1)
    const rc = try self.resizeFn(self, old_byte_slice, Slice.alignment, new_byte_count, 0, @returnAddress());
               ^
/home/pancake/Documents/repos/zig/lib/std/process.zig:177:46: 0x55bb1e3e7501 in std.process.getEnvVarOwned (zig1)
        const result = os.getenv(key) orelse return error.EnvironmentVariableNotFound;
                                             ^
/home/pancake/Documents/repos/zig/lib/std/os.zig:2159:19: 0x55bb1e37a39b in std.os.mkdiratZ (zig1)
        EEXIST => return error.PathAlreadyExists,
                  ^
/home/pancake/Documents/repos/zig/lib/std/os.zig:2120:9: 0x55bb1e37a256 in std.os.mkdirat (zig1)
        return mkdiratZ(dir_fd, &sub_dir_path_c, mode);
        ^
/home/pancake/Documents/repos/zig/lib/std/fs.zig:1044:9: 0x55bb1e37a166 in std.fs.Dir.makeDir (zig1)
        try os.mkdirat(self.fd, sub_path, default_new_dir_mode);
        ^
/home/pancake/Documents/repos/zig/lib/std/os.zig:2159:19: 0x55bb1e37a39b in std.os.mkdiratZ (zig1)
        EEXIST => return error.PathAlreadyExists,
                  ^
/home/pancake/Documents/repos/zig/lib/std/os.zig:2120:9: 0x55bb1e37a256 in std.os.mkdirat (zig1)
        return mkdiratZ(dir_fd, &sub_dir_path_c, mode);
        ^
/home/pancake/Documents/repos/zig/lib/std/fs.zig:1044:9: 0x55bb1e37a166 in std.fs.Dir.makeDir (zig1)
        try os.mkdirat(self.fd, sub_path, default_new_dir_mode);
        ^
/home/pancake/Documents/repos/zig/lib/std/os.zig:1225:24: 0x55bb1e586ad2 in std.os.openatZ (zig1)
            ENOTDIR => return error.NotDir,
                       ^
/home/pancake/Documents/repos/zig/lib/std/fs.zig:1340:25: 0x55bb1e587b12 in std.fs.Dir.openDirFlagsZ (zig1)
            else => |e| return e,
                        ^
/home/pancake/Documents/repos/zig/lib/std/fs.zig:1310:13: 0x55bb1e37a4f9 in std.fs.Dir.openDirZ (zig1)
            return self.openDirFlagsZ(sub_path_c, os.O_DIRECTORY | os.O_RDONLY | os.O_CLOEXEC | O_PATH | symlink_flags);
            ^
/home/pancake/Documents/repos/zig/lib/std/fs.zig:1255:13: 0x55bb1e37a0fe in std.fs.Dir.openDir (zig1)
            return self.openDirZ(&sub_path_c, args);
            ^
/home/pancake/Documents/repos/zig/lib/std/fs.zig:1095:9: 0x55bb1e36d1bb in std.fs.Dir.makeOpenPath (zig1)
        return self.openDir(sub_path, open_dir_options);
        ^
/home/pancake/Documents/repos/zig/src/main.zig:1780:25: 0x55bb1e30e35d in main.buildOutputType (zig1)
            const dir = try fs.cwd().makeOpenPath(local_cache_dir_path, .{});
                        ^
/home/pancake/Documents/repos/zig/src/main.zig:178:9: 0x55bb1e2e5da7 in main.mainArgs (zig1)
        return buildOutputType(gpa, arena, args, .run);
        ^
???:?:?: 0x55bb1e2e70fb in ??? (???)
/home/pancake/Documents/repos/zig/src/stage1.zig:45:43: 0x55bb1e2e55a1 in main (zig1)
        stage2.mainArgs(gpa, arena, args) catch unreachable;
                                          ^
???:?:?: 0x7f958f8930b2 in ??? (???)
```

**Comment by linkpy** [comment]
A bit more investigation : ZLS runs that command when trying to get the package list : 

```
/home/pancake/Binaries/zig run /home/pancake/Documents/repos/zls/zig-out/bin/build_runner.zig --cache-dir /home/pancake/Documents/repos/zls/zig-out/bin/build_runner.zig --pkg-begin @build@ /home/pancake/Documents/repos/zls/build.zig --pkg-end
```

Zig panics because the cache directory is the build_runner.zig file.

**Comment by alexnask** [comment]
Sorry about this, it seems like I never tested manually setting the build_runner path >.>
Should be closed in 14dc9d1b51538fb0b51ed244b4e9db9415ee79ed, please reopen if you face another issue with this.  
Thanks for your patience

**Comment by linkpy** [comment]
It works perfectly! Thank you.
